src/components/UserProfile/UpdateProfile/UpdateProfile.js

```diff
@@ -0,0 +1,205 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { ClipLoader } from 'react-spinners';
+import { withToastManager } from 'react-toast-notifications';
+import FormContainer from '../../../containers/Form/FormContainer';
+import { Button, Input } from '../../utilities';
+import actionCreators from '../../../actions/userAction/updateProfile';
+import './_updateProfile.scss';
+
+
+/**
+ * @class UpdateProfile
+ * @description The class represent the update profile logic
+ */
+export class UpdateProfile extends Component {
+  /**
+   * @param {object} props - this.props
+   * @constructor
+   * @description holds the state;
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      facebook: this.props.userData.facebook,
+      twitter: this.props.userData.twitter,
+      bio: this.props.userData.bio,
+      firstName: this.props.userData.firstName,
+      lastName: this.props.userData.lastName,
+      email: this.props.userData.email,
+      userName: this.props.userData.userName,
+      image: this.props.userData.image,
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
```

Comment:
Hi Michael, Great Work.
However, you are adviced to use arrow functions when creating your methods so you don't need to bind `this` at the constructor. You also don't need to create state in your constructor. you can create a state property and it will work. We already set up babel to handle it.
```
class Michael extend Component{
  state = {
    facebook: this.props.userData.facebook,
  }
  onSubmit = () => {
    //expression
  }
}

```

```

```

Reply:
@JohnMadakin Thanks for the feedback.